tests/integration/test_charm.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+import lightkube
+from lightkube.generic_resource import create_global_resource
+
+log = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+
+APP_NAME = "kfserving"
+CHARM_ROOT = "."
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    # build charm
+    charm_under_test = await ops_test.build_charm(CHARM_ROOT)
+    log.info(f"Built charm {charm_under_test}")
+
+    # deploy charm
+    image_path = METADATA["resources"]["oci-image"]["upstream-source"]
+    resources = {"oci-image": image_path}
+    await ops_test.model.deploy(charm_under_test, resources=resources)
+
+    # wait until active and idle
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60 * 10
+    )
+
+    # check status
+    charm_status = ops_test.model.applications[APP_NAME].units[0].workload_status
+    log.info(f"charm {charm_under_test} status is {charm_status}")
+    assert charm_status == "active"
+
+
+async def test_flower_predictor(ops_test: OpsTest, lightkube_client):
```

Comment:
This might just be an opinion, but we either remove the test case or skip it with an appropriate skip message, to prevent false positives. Pytest has a skip [implementation](https://docs.pytest.org/en/latest/how-to/skipping.html#skipping-test-functions).

Reply:
Didn't know about this! TIL thanks Daniela :)